File: app/src/main/java/ua/leonidius/rtlnotepad/dialogs/LoadingDialog.java
package ua.leonidius.rtlnotepad.dialogs;

import android.app.Dialog;
import android.app.ProgressDialog;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;
import ua.leonidius.rtlnotepad.R;

public class LoadingDialog extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        ProgressDialog.Builder pdb = new ProgressDialog.Builder(getActivity());
        pdb.setMessage(R.string.processing);
        pdb.setCancelable(false);
        return pdb.create();
    }

}

File: app/src/main/java/ua/leonidius/rtlnotepad/dialogs/LoadingDialog.kt
package ua.leonidius.rtlnotepad.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.util.Log
import androidx.fragment.app.DialogFragment
import ua.leonidius.rtlnotepad.R

class LoadingDialog : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        Log.d("LoadingDialog", "Inside onCreateDialog")
        val pdb = AlertDialog.Builder(activity)
        pdb.setTitle(R.string.processing)
        pdb.setMessage(R.string.processing)
        pdb.setCancelable(false)
        return pdb.create()
    }

}